src/api/baseapi.cpp

```diff
@@ -1034,11 +1034,22 @@ bool TessBaseAPI::ProcessPagesMultipageTiff(const l_uint8 *data,
   int page = (tessedit_page_number >= 0) ? tessedit_page_number : 0;
   size_t offset = 0;
   for (; ; ++page) {
-    if (tessedit_page_number >= 0)
+    if (tessedit_page_number >= 0) {
       page = tessedit_page_number;
-    pix = (data) ? pixReadMemFromMultipageTiff(data, size, &offset)
-                 : pixReadFromMultipageTiff(filename, &offset);
-    if (pix == nullptr) break;
+      int pages_read = 0;
+      do {
+        pix = (data) ? pixReadMemFromMultipageTiff(data, size, &offset)
+                     : pixReadFromMultipageTiff(filename, &offset);			  
+        pages_read++;
+      } while (pix != nullptr && pages_read < (page + 1));
+    }
+    else {
```

Comment:
This is not the Tesseract coding style.

Reply:
There could be set automatic style check based on clang format. Many projects have such thing.

Reply:
While thinking more about that code and after reading the discussion on issue #233, I now think more changes would be reasonable. The new code loops reading a pix for each page until it reaches the desired page. It won't stop the loop at the end of the TIFF file if the user gave a too large page number, but starts again with reading the first page. The read data is never released, so it is easy to get out of memory:

    $ time bin/debug/x86_64-linux-gnu/src/api/tesseract issue/2537/18.TIF - -c tessedit_page_number=1000000
    Error in pixCreateNoInit: pix_malloc fail for data
    Error in pixCreate: pixd not made
    Error in pixReadFromTiffStream: pix not made

    real    1m26,521s
    user    0m26,758s
    sys     0m13,830s

In the past, `pixReadTiff` was used to read the n-th image. I think that is better than calling `pixReadFromMultipageTiff` in a loop, but I am not sure why it was replaced for the case of a single image (for handling all or a sequence of images, `pixReadFromMultipageTiff` is superior).

Reply:
> This is not the Tesseract coding style.

i will careful when next commit.